pseudomonas_analysis/nbconverted/explore_expression_data.py

```diff
@@ -0,0 +1,634 @@
+)
+
+
+# In[19]:
+
+
+# Compare the distribution of adjusted p-value scores
+# for the volcano plot that doesn't level off as much
+# as we increase logFC
+simulated_DE_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_simulated_data_{project_id}_2.txt"
+)
+
+simulated_DE = pd.read_csv(
+    simulated_DE_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+
+# In[20]:
+
+
+sns.distplot(leveloff_simulated_DE[pval_name])
+sns.distplot(simulated_DE[pval_name])
+
+
+# In[21]:
+
+
+# Get genes in peak of distribution (blue)
+gene_ids = list(leveloff_simulated_DE[leveloff_simulated_DE[pval_name]<0.4].index)
+print(leveloff_simulated_DE[leveloff_simulated_DE[pval_name]<0.4].head(20))
+
+leveloff_simulated_expression_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_1.txt"
+)
+
+leveloff_simulated_expression = pd.read_csv(
+    leveloff_simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+sns.heatmap(leveloff_simulated_expression[gene_ids[0:20]])
+
+
+# In[22]:
+
+
+# Get genes in peak of distribution (orange)
+gene_ids = list(simulated_DE[simulated_DE[pval_name]<0.4].index)
+print(simulated_DE[simulated_DE[pval_name]<0.4].head(20))
+
+simulated_expression_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_2.txt"
+)
+
+simulated_expression = pd.read_csv(
+    simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+sns.heatmap(simulated_expression[gene_ids[0:20]])
+
+
+# In[23]:
+
+
+# Let's look at the distribution of adjusted p-value scores for those volcano
+# plots that is completely flat
+simulated_DE_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_simulated_data_{project_id}_3.txt"
+)
+
+flat_simulated_DE = pd.read_csv(
+    simulated_DE_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+
+# In[24]:
+
+
+sns.distplot(flat_simulated_DE[pval_name])
+
+
+# In[25]:
+
+
+sns.distplot(simulated_DE[pval_name])
+
+
+# In[26]:
+
+
+# Get genes in peak of distribution (flat)
+gene_ids = list(flat_simulated_DE.index)
+print(flat_simulated_DE.head(20))
+
+flat_simulated_expression_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_3.txt"
+)
+
+flat_simulated_expression = pd.read_csv(
+    flat_simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+sns.heatmap(flat_simulated_expression[gene_ids[0:20]])
+
+
+# Looks like there is a peak of adjusted p-values at the minimum range of the distribution which is causing the leveling out (i.e. there are many genes with a similar low adjusted p-value). I am guessing this is also a result of the VAE shrinkage, where instead of genes having varying logFC, genes are compressed such that there are groups of genes with similar logFC and therefore similar adjusted p-values.
```

Comment:
Yeah, this would be my guess too based on your analysis. The samples you chose in the peak really seem to have strikingly similar gene expression in your heatmap, at least for the genes you chose. Interesting!